File: java/src/algorithms/warmup/PlusMinus.java
package algorithms.warmup;

import java.util.*;
import java.util.stream.*;

public class PlusMinus {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        in.nextLine();
        String input = in.nextLine();
        int[] a = Stream.of(input.split(" ")).mapToInt(Integer::parseInt).toArray();

        int pos = 0, neg = 0, zero = 0;
        for (int x : a) {
            if (x > 0) {
                pos++;
            }
            else if (x < 0) {
                neg++;
            }
            else {
                zero++;
            }
        }

        System.out.println((float) pos / n);
        System.out.println((float) neg / n);
        System.out.println((float) zero / n);
    }
}

File: python/algorithms/warmup/plus_minus.py
n = int(input().strip())
a = [int(x) for x in input().strip().split(' ')]

pos, neg, zero = 0, 0, 0
for x in a:
    if x > 0:
        pos += 1
    elif x < 0:
        neg += 1
    else:
        zero += 1

print(float(pos) / n)
print(float(neg) / n)
print(float(zero) / n)
